docs/src/modules/spring/pages/development-process-spring.adoc

```diff
@@ -1 +1,86 @@
+. <<_package_service>>
+. <<_run_locally>>
+. <<_deploy_to_kalix>>
+
+
+[#_specify_service_interface_and_domain_model]
+== Specify service interface and domain model
+
+The main components of a Kalix service are:
+
+* Stateful https://docs.kalix.io/reference/glossary.html#entity[Entities]
+* Stateless https://docs.kalix.io/reference/glossary.html#action[Actions]
+* https://docs.kalix.io/reference/glossary.html#view[Views], which return the state of multiple instances of a stateful entity.
+
+We recommend that you separate the service API and Entity domain data model. Separating the service interface and data model in different classes allows you to evolve them independently.
+
+TIP: Kickstart a project using the xref:spring:getting-started.adoc[Getting started] guide.
+
+
+[#_implement_components]
+== Implement components
+
+Stateful services can implement https://docs.kalix.io/reference/glossary.html#value_entity[Value Entity] or https://docs.kalix.io/reference/glossary.html#event_sourced_entity[Event Sourced Entity] and https://docs.kalix.io/reference/glossary.html#view[View] components. Stateless services implement https://docs.kalix.io/reference/glossary.html#action[Actions]. Typically, a stateful service should only have one Entity type, but can also include Actions and Views.
+
+=== Actions
+
+Actions are stateless functions that can be triggered by HTTP calls. They can also subscribe to published events, and they can send commands to other services based on the events they receive. For more information about writing actions see xref:spring:actions.adoc[Implementing Actions].
+
+=== Entities
+Stateful services encapsulate business logic in Value Entities or Event Sourced Entities. At runtime, command messages invoke operations on Entities. A command may only act on one Entity at a time.
+
+
+TIP: To learn more about Kalix entities see xref:spring:value-entity.adoc[Implementing Value Entities], xref:spring:eventsourced.adoc[Implementing Event Sourced Entities], and xref:spring:replicated-entity.adoc[Implementing Replicated Entities].
+
+If you would like to update multiple Entities from a single command, you can use forwarding and side effects.
+
+TIP: For more information see xref:spring:actions-as-controller.adoc[Actions as Controllers] and xref:spring:side-effects.adoc[Running Side Effects].
+
+Services can interact asynchronously with other services and with external systems. Event Sourced Entities emit events to a journal, to which other services can subscribe. By configuring your own publish/subscribe (pub/sub) mechanism, any service can publish their own events and subscribe to events published by other services or external systems.
+
+For more details and examples take a look at the following topics:
+
+* xref:spring:value-entity.adoc[Value Entities]
+* xref:spring:eventsourced.adoc[Event Sourced Entities]
+* xref:spring:replicated-entity.adoc[Replicated Entities]
+* xref:spring:actions.adoc[Actions]
+* xref:spring:views.adoc[Views]
```

Comment:
```suggestion
```

Don't think this list adds much since the first 4 links were just mentioned in the previous paragraphs (in very highlighted ways).. The only one missing is for Views, and there I would probably replace this with a small section for `Views` where you briefly explain what are they used for and link to the detail page, similar to what we do for Actions and Entities. A bit more consistent and less duplicated links.

Reply:
Good point